bytesconv.go

```diff
@@ -363,6 +361,21 @@ func lowercaseBytes(b []byte) {
 	}
 }
 
+// fastlen for byte slices uses slice len field directly
+// since it's pretty much safe, it will break only if slice
+// implementation will change, that seems to be impossible
+// without releasing Go2
+//
+// here we taking slice pointer, converting it to unsafe.Pointer
+// instruction, then converting it to a pointer to array of three
+// unsafe.Pointers, reading the middle one (1st index) as a uintptr,
+// then converting it to int. cause we don't have a real unsafe.Pointer
+// conversions at runtime (there is no real unsafe.Pointer type),
+// it's cheap and fast
+func fastlen(b []byte) int {
```

Comment:
But it's not, I've tested it on latest versions and fastlen still faster than len()